rlp/sedes/serializable.py

```diff
@@ -0,0 +1,299 @@
+    if unknown_kwargs:
+        raise TypeError("Unknown kwargs: {0}".format(sorted(unknown_kwargs)))
+
+    missing_kwargs = set(needed_kwargs).difference(kwargs.keys())
+    if missing_kwargs:
+        raise TypeError("Missing kwargs: {0}".format(sorted(missing_kwargs)))
+
+    yield from args
+    for arg_name in needed_kwargs:
+        yield kwargs[arg_name]
+
+
+class BaseSerializable(collections.Sequence):
+    @property
+    @abc.abstractmethod
+    def _cached_rlp(self):
+        pass
+
+    def __init__(self, *args, **kwargs):
+        field_values = merge_args_and_kwargs(args, kwargs, self._meta.field_names)
+        if field_values:
+            field_attrs = self._get_field_attrs()
+
+            if len(field_values) != len(field_attrs):
+                raise TypeError('Argument count mismatch. expected {0} - got {1}'.format(
+                    len(field_attrs),
+                    len(field_values),
+                ))
+            for value, attr in zip(field_values, field_attrs):
+                setattr(self, attr, value)
+
+    def __iter__(self):
+        for field in self._meta.get_field_names():
+            yield getattr(self, field)
+
+    def __getitem__(self, idx):
+        if isinstance(idx, int):
+            field = self._meta.field_names[idx]
+            return getattr(self, field)
+        else:
+            raise NotImplementedError("not implemented")
+
+    def __len__(self):
+        return len(self._meta.fields)
+
+    def __eq__(self, other):
+        if not hasattr(other, 'serialize'):
+            return False
+        return self.serialize(self) == other.serialize(other)
+
+    @classmethod
+    def get_sedes(cls):
+        return cls._meta.sedes
+
+    @classmethod
+    def serialize(cls, obj):
+        try:
+            result = cls.get_sedes().serialize(obj)
+        except ListSerializationError as e:
+            raise ObjectSerializationError(obj=obj, sedes=cls, list_exception=e)
+        else:
+            return result
+
+    @classmethod
+    def deserialize(cls, serial, exclude=None, mutable=False, **kwargs):
+        try:
+            values = cls.get_sedes().deserialize(serial)
+        except ListDeserializationError as e:
+            raise ObjectDeserializationError(serial=serial, sedes=cls, list_exception=e)
+
+        params = {
+            field: value
+            for field, value
+            in zip(cls._meta.field_names, values)
+            if exclude is None or field not in exclude
+        }
+        obj = cls(**dict(list(params.items()) + list(kwargs.items())))
+
+        # TODO: mutability
+        if mutable:
+            return obj.as_mutable()
+        else:
+            return obj.as_immutable()
+
+    @classmethod
+    def exclude(cls, excluded_fields):
+        """Create a new sedes considering only a reduced set of fields."""
+        raise NotImplementedError("Not yet implemented")
+
+    @property
+    def is_immutable(self):
```

Comment:
We probably don't need this as we can just inline the `not self.is_mutable`, no?